compiler/rustc_typeck/src/check/method/prelude2021.rs

```diff
@@ -353,12 +359,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             Some(probe::AutorefOrPtrAdjustment::ToConstPtr) | None => "",
         };
 
-        let (expr_text, precise) =
-            if let Ok(expr_text) = self.sess().source_map().span_to_snippet(expr.span) {
-                (expr_text, true)
-            } else {
-                ("(..)".to_string(), false)
-            };
+        let (expr_text, precise) = if let Ok(expr_text) = self
+            .sess()
+            .source_map()
+            .span_to_snippet(expr.span.find_ancestor_inside(outer).unwrap_or_default())
```

Comment:
Same. It feels like it should be `expr.span.find_ancestor_inside(outer).and_then(...`